### PR TITLE
feat: Add "local" source manager

### PR DESCRIPTION
**Pull request by masci**
This PR adds a local source manager that can be used instead of the git source manager for local experiments and demonstration purposes. 

It was used to rewrite the README and can be possibly used in the docs as well, see the preview [here](https://github.com/run-llama/llama_deploy/blob/massi/local-source/README.md).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,56 +1,183 @@
...
+_Control Plane_ handles ongoing tasks, manages the internal state, keeps track of which services are available, and
+decides which service should handle the next step of a task using another internal component called _Orchestrator_.
+A well defined set of these components is called _Deployment_, and a single Llama Deploy instance can serve multiple
+of them.
 
-At the top of a `llama_deploy` system is the `control plane`. The control plane handles ongoing tasks, manages state, keeps track of which services are in the network, and also decides which service should handle the next step of a task using an `orchestrator`. The default `orchestrator` is purely programmatic, handling failures, retries, and state-passing.
+The goal of Llama Deploy is to easily transition something that you built in a notebook to something running on the
+cloud with the minimum amount of changes to the original code, possibly zero. In order to make this transition a
+pleasant one, the intrinsic complexity of running agents as services is managed by a component called _API Server_,
+the only one in LLama Deploy that's user facing. You can interact with the API Server in two ways:
```
Comment (nerdai) [inline README.md]: nit: "Llama Deploy"

**Review by nerdai — APPROVED** [review]
looks really good!

small nit: A couple of instances of "LLama Deploy" in the README. A quick search and replace to fix would be great.

File: README.md [hunk 2]
```diff
@@ -1,56 +1,183 @@
...
+    # We tell Llama Deploy where to look for our workflow
+    source:
+      # In this case, we instruct Llama Deploy to look in the local filesystem
+      type: local
+      # The path in the local filesystem where to look. This assumes there's an src folder in the
+      # current working directory containing the file workflow.py we created previously
+      name: ./src
+    # This assumes the file workflow.py contains a variable called `echo_workflow` containing our workflow instance
+    path: workflow:echo_workflow
+```
```
Comment (logan-markewich) [inline README.md:118]: This makes me want `llamactl init` 😁

**Review by logan-markewich — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (masci) [reply README.md:118]: Let's do that!

Comment (masci) [reply README.md:118]: https://github.com/run-llama/llama_deploy/issues/287

**Pull request merged**